### PR TITLE
fix: implement InstanceDisposer to close db connections

### LINKED ISSUE #135: Database connections are not closed when a new datasource is created
`grafana-plugin-sdk-go` supports a `Dispose()` method to be called when a [new Instance is to be created](https://github.com/grafana/grafana-plugin-sdk-go/blob/5b4c6d330e527763e5f2373e56630606ed3dc8bc/backend/instancemgmt/instance_manager.go#L123-L127). Sqlds does not currently support this. This leaks database connections in some circumstances.

`SQLDatasource` should support the `InstanceDisposer` interface and have its `Connector` close its open database connections when `Dispose()` is called. I'm working on a PR for this now.

### PR DESCRIPTION
**Pull request by njvrzm**
This adds support for the `InstanceDisposer()` interface to `SQLDatasource`, to fix a connection leak.

Fixes #135

### CODE REVIEW

File: datasource_connect_test.go [hunk 1]
```diff
@@ -108,20 +119,21 @@ func Test_getDBConnectionFromQuery(t *testing.T) {
...
+		conn.connections.Store("foo", dbConnection{db: db})
 		ds.Dispose()
 		count := 0
 		conn.connections.Range(func(key, value interface{}) bool {
 			count++
 			return true
 		})
-		if count != 2 {
-			t.Errorf("missing connections")
+		if count != 0 {
```
Comment (iwysiu) [inline datasource_connect_test.go:135]: Do we know why this was originally checking that we don't delete connections?

**Review by iwysiu — APPROVED** [review]

File: datasource_connect_test.go (hunk 1 above)
Comment (njvrzm) [reply datasource_connect_test.go:135]: I don't! Except that, of course, it originally didn't. I suppose it's possible that existing datasources are relying on that behavior, if their connectors reuse connections? But I feel like that would be a bit weird, and it doesn't look like any of ours do that.

You could argue that this should be a new major version, but to me this feels more like a bugfix than a backwards incompatible change, even if the bug was originally intentional. If that makes sense.

Comment (njvrzm) [reply datasource_connect_test.go:135]: (Fwiw I looked over the datasources - grafana and non-grafana - using `sqlds.NewDatasource` and it doesn't look like any of them reuse connections this way.)

**Review by aangelisc — APPROVED** [review]
The change looks good to me. I'm a little surprised that we weren't actually closing the connections before 🤔 I would suggest releasing this as a major because it is a functionality change. I agree that it does seem like a bugfix also but we can't guarantee that any community plugins (if there are any) based on SQLds don't reuse connections in an unexpected fashion.

**Comment by njvrzm** [comment]
Fair enough - I looked over the github projects using it (only a handful outside of grafana) but github isn't the whole universe. I feel a little weird bumping to v5 just a couple months after v4, but so it goes :)

**Pull request merged**